<a href="https://colab.research.google.com/github/anindyaa25/Project-KKA-NabilaAnindya/blob/main/data_kantin_NabilaAnindya24.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [12]:
#Latihan 1: Operasi Dasar NumPy Array

import numpy as np

harga = np.array([5000, 7000, 3000, 12000, 4500])
print('Rata-rata harga:', harga.mean())
print('Harga tertinggi:', harga.max())
print('Harga setelah diskon 10%:', harga * 0.9)


Rata-rata harga: 6300.0
Harga tertinggi: 12000
Harga setelah diskon 10%: [ 4500.  6300.  2700. 10800.  4050.]


Tugas Analisis 1:
 Bandingkan hasil harga * 0.9 di atas dengan jika harga berupa list Python  biasa (bukan array). Apa yang terjadi bila kamu mencoba [5000,7000,3000] * 0.9? Jelaskan  mengapa berbeda.

= Jika menggunakan list python biasa Akan Error karena operator * berarti mengulang atau menggandakan list, bukan mengalikan tiap elemen.

In [13]:
#Latihan 2

import pandas as pd

data_kantin = {
 'menu': ['Nasi Goreng', 'Es Teh', 'Mie Ayam', 'Es Teh', None],
 'harga': [12000, 4000, 10000, 4000, 8000],
 'terjual': [23, 40, None, 35, 18]
}
df = pd.DataFrame(data_kantin)
print(df)


          menu  harga  terjual
0  Nasi Goreng  12000     23.0
1       Es Teh   4000     40.0
2     Mie Ayam  10000      NaN
3       Es Teh   4000     35.0
4         None   8000     18.0


Tugas Analisis 2:

Amati tabel yang dihasilkan. Kolom mana yang terlihat memiliki data kosong  (None)? Menurutmu apa risikonya jika data ini langsung dianalisis tanpa dibersihkan?

= Ada 2 kolom yang menghasilkan data kosong yaitu menu (baris ke 5), terjual (baris ke 3). Resikonya perhitungan dan analisis jadi tidak lengkap dan tidak akurat.


In [ ]:
#LATIHAN 3
df = pd.read_csv('data_kantin.csv')

print(df.head()) #5 baris pertama
print(df.info()) # tipe data dan jumlah non-null tiap kolom
print(df.describe()) # statistik ringkas kolom numerik
print(df.shape) # jumlah (baris, kolom)




      tanggal         menu kategori  harga  terjual
0  2026-08-20   Teh Hangat  Minuman   3000     11.0
1  2026-08-31  Nasi Goreng  Makanan  12000     29.0
2  2026-08-21  Nasi Goreng  Makanan  12000     28.0
3  2026-08-28   Teh Hangat  Minuman   3000     12.0
4  2026-08-21     Es Jeruk  Minuman   5000     19.0
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 110 entries, 0 to 109
Data columns (total 5 columns):
 #   Column    Non-Null Count  Dtype  
---  ------    --------------  -----  
 0   tanggal   110 non-null    object 
 1   menu      107 non-null    object 
 2   kategori  110 non-null    object 
 3   harga     110 non-null    int64  
 4   terjual   104 non-null    float64
dtypes: float64(1), int64(1), object(3)
memory usage: 4.4+ KB
None
              harga     terjual
count    110.000000  104.000000
mean    6336.363636   22.923077
std     3520.001232   10.831178
min     2000.000000    8.000000
25%     3000.000000   13.000000
50%     5000.000000   20.000000
75%    10000.000000 

**Tugas Analisis 3:**
Dari hasil df.info(), kolom mana yang jumlah non-null-nya lebih sedikit dari  jumlah baris total? Apa artinya?    

= Baris total nya ada 110 dan di bagian menu (107  non-null), terjual (104 non-null).
Artinya, kolom menu kekurangan 3 data dan kolom terjual kekurangan 6 data. Data yang hilang ini adalah missing value (NaN/None).

In [ ]:
#Latihan 4: Menangani Missing Value

print(df.isnull().sum()) # jumlah data kosong tiap kolom
df['terjual'] = df['terjual'].fillna(0) # isi kekosongan dengan 0
df = df.dropna(subset=['menu']) # hapus baris jika kolom menu kosong


tanggal     0
menu        3
kategori    0
harga       0
terjual     6
dtype: int64


Tugas Analisis 4:

Mengapa pada contoh di atas kolom terjual diisi (fillna) sedangkan baris  dengan menu kosong justru dihapus (dropna)? Diskusikan alasan logisnya.

= Karena jumlah_terjual adalah data yang isinya angka, jadi masih bisa ditebak nilainya.
Fillna itu masih bisa diperkirakan dan baris lainnya masih berguna, sementara dropna itu gabisa ditebak jadi ga berguna kalau di pertahankan.


In [ ]:
#Latihan 5: Menangani Duplikat dan Tipe Data

print(df.duplicated().sum()) # jumlah baris duplikat

df = df.drop_duplicates()
df['harga'] = df['harga'].astype(int) # memastikan tipe data harga adalah integer

print(df.dtypes)

print("Jumlah baris sebelum:", df.shape[0])
df = df.drop_duplicates()
print("Jumlah baris sesudah:", df.shape[0])


0
menu        object
harga        int64
terjual    float64
dtype: object
Jumlah baris sebelum: 5
Jumlah baris sesudah: 5


Tugas Analisis 5:

Setelah drop_duplicates(), berapa jumlah baris dataset sekarang dibanding  sebelumnya? Mengapa penting memastikan tipe data (dtypes) sudah benar sebelum
data dianalisis lebih lanjut?

= awalnya ada 5 baris duplikat di dataset, duplikast itu di hapus pakai drop_duplicates() makanya pas di cek ulang di akhir, jumlah baris sebelum dan sesudah sama sama 102- itu karena duplikatnya sudah terlanjur dihapus duluan, bukan karena memang dari awal nggak ada duplikat.

In [ ]:
#Latihan 6: Data Manipulation (Filtering, Sorting, Groupby)

laris = df[df['terjual'] > 20] # filtering
urut = df.sort_values(by='terjual', ascending=False) # sorting

df['total_pendapatan'] = df['harga'] * df['terjual'] # kolom turunan

ringkasan = df.groupby('menu')['total_pendapatan'].sum() # agregasi
print(ringkasan)


menu
Es Jeruk       1460000.0
Es Teh         2144000.0
Kerupuk         972000.0
Mie Ayam       2410000.0
Nasi Goreng    3924000.0
Roti Bakar     1288000.0
Teh Hangat      459000.0
Name: total_pendapatan, dtype: float64


Tugas Analisis 6:

 Dari hasil groupby di atas, menu apa yang menghasilkan total_pendapatan tertinggi? Bagaimana informasi ini bisa membantu pengambilan keputusan di kantin sekolah?

 = jadi menu dengan  penghasilan_tertinggi adalah nasi goreng, informasi ini bisa membantu mengetahui untuk membuat strategi promosi, evaluasi menu yang kurang diminati di kantin sekolah. informasi ini bisa jadi bahan pertimbangan untuk membantu pengambilan keputusan di kntin sekolah.